libs/application/ui-shell/src/reducer/reducerUtils.ts

```diff
@@ -49,10 +56,25 @@ export const findCurrentScreen = (
       if (getValueViaPath(answers, screen.id) !== undefined) {
         currentScreen = index
       }
-    } else if (screen.id && getValueViaPath(answers, screen.id) !== undefined) {
-      currentScreen = index + 1
+    } else if (screen.id) {
+      if (
+        getValueViaPath(answers, screen.id) === undefined &&
+        index > currentAnswerIndex
+      ) {
```

Comment:
Should we change `screens.forEach` to a `for` loop and `break` on this condition?

Reply:
Then we can skip the `!missingAnswerBeforeCurrentIndex` below

Reply:
Yes, agree on that. However, I will keep that for another PR, I don't want to widespread too much on this one